src/Controls/src/Core/Accelerator.cs

```diff
@@ -1,42 +1,62 @@
-#nullable disable
+﻿#nullable disable
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Linq;
 
 namespace Microsoft.Maui.Controls
 {
 	/// <include file="../../docs/Microsoft.Maui.Controls/Accelerator.xml" path="Type[@FullName='Microsoft.Maui.Controls.Accelerator']/Docs/*" />
 	[System.ComponentModel.TypeConverter(typeof(AcceleratorTypeConverter))]
-	public class Accelerator
+	public class Accelerator : IAccelerator
 	{
-		const char Separator = '+';
-		string _text;
+		const string Separator = "+";
+		readonly string _text;
 
-		internal Accelerator(string text)
+		internal Accelerator(string text, IEnumerable<string> modifiers, string key)
 		{
 			if (string.IsNullOrEmpty(text))
 				throw new ArgumentNullException(nameof(text));
+
 			_text = text;
+			Key = key;
+			Modifiers = modifiers;
 		}
 
-		/// <include file="../../docs/Microsoft.Maui.Controls/Accelerator.xml" path="//Member[@MemberName='Modifiers']/Docs/*" />
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		public IEnumerable<string> Modifiers { get; set; }
+		/// <summary>
+		/// Gets the modifiers for the accelerator.
+		/// </summary>
+		public IEnumerable<string> Modifiers { get; }
+
+		IReadOnlyList<string> IAccelerator.Modifiers => Modifiers?.ToList();
```

Comment:
I _think_ this could be simplified and we could remove the performance hit of `ToList()` on every access by storing the modifiers as a `List<string>`. Something like this:

```csharp
readonly List<string> _modifiers;

internal Accelerator(string text, IEnumerable<string> modifiers, string key)
{
	...
	_modifiers = new List<string>(modifiers);
}
	
public IEnumerable<string> Modifiers => _modifiers;
IReadOnlyList<string> IAccelerator.Modifiers => _modifiers;
```

Then we could drop that `using System.Linq;` line, too.

Reply:
Nice suggestion, changed.